docs/design/partitioning/07-parallel-migrations.md

```diff
@@ -0,0 +1,536 @@
+native.memory.max=495.0G, 
+native.meta.memory.used=1011.8M, 
+native.meta.memory.free=4.0G
+```
+
+**Native memory usage with 9 members:**
+```
+native.memory.used=460.4G, 
+native.memory.free=34.6G, 
+native.memory.total=460.7G, 
+native.memory.max=495.0G, 
+native.meta.memory.used=1.1G, 
+native.meta.memory.free=3.9G
+```
+
+This was total of `~4.1 TB` data (including the backup) in the whole cluster.
+
+### Test Flow
+
+```shell script
+# Initially started 10 members
+hz cluster -size M10 -v 4.1-SNAPSHOT -ee -boxes a.box -user ec2-user
+
+# Load initial data into map
+hz run ops/load     
+# Validate map size
+hz run ops/mapSize
+               
+# Start get/set ops at the background
+hz run ops/get &
+hz run ops/set &
+       
+# Scale-down 
+# Shutdown Member10
+hz driver Member10
+hz run ops/shutDown            
+# Verify cluster-size=9 and wait until migrations complete
+hz run ops/untilClusterSize9 ops/untilClusterSafe        
+
+# Scale-up
+# Start Member10 back
+hz boot Member10
+# Verify cluster-size=10 and wait until migrations complete
+hz run ops/untilClusterSize10 ops/untilClusterSafe
+
+# Member failure
+# Kill Member10
+hz kill Member10
+# Verify cluster-size=9 and wait until migrations complete
+hz run ops/untilClusterSize9 ops/untilClusterSafe9
+```
+
+### Single Map Benchmark
+
+Used a single HD `IMap` and inserted `135 million` entries with `~16k` values. 
+
+#### Master Branch
+
+- *Scale down from 10 to 9 members*: `14000` migrations were completed in **35 minutes and 34 seconds**. 
+Member shutdown in **2145 seconds**.
+
+- *Scale up from 9 to 10 members*: `14000` migrations were completed in **34 minutes and 16 seconds**.
+
+- *Crash single member*: Around `~26150` migrations were completed in **33 minutes and 38 seconds**.
+
+| GET p99 Latency  | GET p99.9 Latency |
+|:-------------:|:-------------:|
+| ![Plot](resources/07/single/master/get/p99.png) | ![Plot](resources/07/single/master/get/p999.png) |
+| GET p99 Min-Max  | GET p99.9 Min-Max |
+| ![Plot](resources/07/single/master/get/p99-min-mean-max.png) | ![Plot](resources/07/single/master/get/p999-min-mean-max.png) |
+
+
+| SET p99 Latency  | SET p99.9 Latency |
+|:-------------:|:-------------:|
+| ![Plot](resources/07/single/master/set/p99.png) | ![Plot](resources/07/single/master/set/p999.png) |
+| SET p99 Min-Max  | SET p99.9 Min-Max |
+| ![Plot](resources/07/single/master/set/p99-min-mean-max.png) | ![Plot](resources/07/single/master/set/p999-min-mean-max.png) |
+
+
+**Heap Usage and GC Pauses**
+
+![Plot](resources/07/single/master/heap.png)
+
+| GC Summary  | |
+|:-------------|-------------:|
+| Full collections |	36 |
+| Mean garbage collection pause (ms) |	89.4 |
+| Mean interval between collections (ms) |	180394 |
+| Proportion of time spent in garbage collection pauses (%) |	3.39 |
+| Proportion of time spent unpaused (%) |	96.61 |
+
+
+---
+
+#### Parallel Migrations - 5x
+
+5x parallelization was used: `hazelcast.partition.max.parallel.migrations=5`.
```

Comment:
I can see no tests with default value 10.

Reply:
Initially I was considering 5 as the default value and during PoC (which you can see the results on wiki) I did tests with 5, 10 and 20. But for final tests I just used 5 and 20 to show there's no very significant difference between them. Then I picked 10 as the default value, so a middle-ground.